subprojects/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -47,6 +47,16 @@ The `groovy-base` plugin is now responsible for configuring source and target co
 If you are using this task *without applying `grooy-base`*, you will have to manually set compatibility versions on these tasks.
 In general, the `groovy-base` plugin should be applied whenever working with Groovy language tasks.
 
+=== Deprecations
+
+[[depending_on_root_configuration]]
+==== Deprecated depending on resolved configuration
+
+In some cases, it is possible to construct a dependency graph where the configuration being resolved can also be selected by a dependency in the graph.
+This can only occur when a configuration is marked as both consumable and resolvable.
+To avoid this, either mark all resolvable configurations as `canBeConsumed=false` or use the `resolvable(String)` configuration factory method.
+In Gradle 9.0, constructing circular graphs in this manner will no longer be allowed and will result in an error.
```

Comment:
🤔 some suggestions...

```suggestion
==== Consuming resolvable configurations as dependencies

When resolving a `Configuration` in a project, it is possible to consume the same configuration as a dependency as well. Configurations should be used for one purpose (resolution, consumption or dependency declarations), so this can only occur when a configuration is marked as both consumable and resolvable.

This can lead to confusing circular dependency graphs or unexpected dependency resolution results. 

To avoid this problem, plugins should mark all resolvable configurations as `canBeConsumed=false` or use the `resolvable(String)` configuration factory method.

In Gradle 9.0, consuming configurations in this manner will no longer be allowed and will result in an error.
```

We should also link to the new Configuration factory API javadoc.